components/dashboard/src/components/RepositoryFinder.tsx

```diff
@@ -0,0 +1,171 @@
+        setSearchQuery(query);
+        await findResults(query, onResults);
+        if (await refreshSearchData(query, user)) {
+            // Re-run search if the underlying search data has changed
+            await findResults(query, onResults);
+        }
+    }
+
+    // Up/Down keyboard navigation between results
+    const onKeyDown = (event: React.KeyboardEvent) => {
+        if (!selectedSearchResult) {
+            return;
+        }
+        const selectedIndex = searchResults.indexOf(selectedSearchResult);
+        const select = (index: number) => {
+            // Implement a true modulus in order to "wrap around" (e.g. `select(-1)` should select the last result)
+            // Source: https://stackoverflow.com/a/4467559/3461173
+            const n = Math.min(searchResults.length, MAX_DISPLAYED_ITEMS);
+            setSelectedSearchResult(searchResults[((index % n) + n) % n]);
+        }
+        if (event.key === 'ArrowDown') {
+            event.preventDefault();
+            select(selectedIndex + 1);
+            return;
+        }
+        if (event.key === 'ArrowUp') {
+            event.preventDefault();
+            select(selectedIndex - 1);
+            return;
+        }
+    }
+
+    useEffect(() => {
+        const element = document.querySelector(`a[href='/#${selectedSearchResult}']`);
+        if (element) {
+            element.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+        }
+    }, [selectedSearchResult]);
+
+    const onSubmit = (event: React.FormEvent) => {
+        event.preventDefault();
+        if (selectedSearchResult) {
+            window.location.href = '/#' + selectedSearchResult;
+        }
+    }
+
+    return <form onSubmit={onSubmit}>
+        <div className="flex px-4 rounded-xl border border-gray-300 dark:border-gray-500">
+            <div className="py-4">
+                <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 16 16" width="16" height="16"><path fill="#A8A29E" d="M6 2a4 4 0 100 8 4 4 0 000-8zM0 6a6 6 0 1110.89 3.477l4.817 4.816a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 010 6z" /></svg>
+            </div>
+            <input type="search" className="flex-grow" placeholder="repository" autoFocus value={searchQuery} onChange={e => search(e.target.value)} onKeyDown={onKeyDown} />
+        </div>
+        <div className="mt-3 -mx-5 px-5 flex flex-col space-y-2 h-64 overflow-y-auto">
+            {searchQuery === '' && searchResults.length === 0 &&
+                <div className="mt-12 mx-auto w-96 text-gray-500">
+                    Paste a <a className="gp-link" href="https://www.gitpod.io/docs/context-urls">repository context URL</a>, or start typing to see suggestions from:
+                    <ul className="list-disc mt-4 pl-7 flex flex-col space-y-1">
+                        <li>Your recent repositories</li>
+                        <li>Your repositories from <Link className="gp-link" to="/integrations">connected integrations</Link></li>
```

Comment:
**issue**: The page redirect when clicking this feels a bit abrupt, no? 😬 

**suggestion**: What if, we skipped linking to the integrations on this empty state and only prompted users to connect more providers when they couldn't find the repository they were looking for? What do you think? Cc @jldec 

| Modal (Empty) | Modal (Missing Repository) |
|-|-|
| <img width="648" alt="1" src="https://user-images.githubusercontent.com/120486/152232865-35eae676-c3c9-40fa-89d1-48dc6b8ef738.png"> | <img width="648" alt="2" src="https://user-images.githubusercontent.com/120486/152232903-408b43f7-4c70-47ac-9bce-54cee43bfd02.png"> |

Reply:
> Hmm, good catch, but I think that's how we agreed to implement links (i.e. no more target="_blank" to open links in a new tab by default).

Correct and agree! To clarify I was _not_ suggesting to open this link in a new tab but either :a: close the modal and redirect to the the integrations page or even better :b: use help links only for external links like docs and skip the link to integrations on the empty state.

Reply:
Thanks! Went with 🅰️ for now. Will open a follow-up issue for the remaining design adjustments.

Reply:
Follow up issue can be found in https://github.com/gitpod-io/gitpod/issues/8088.